src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -279,20 +301,31 @@ internal static void SetProperty(object resource, string propertyName, object va
             }
         }
 
-        internal static void SetInstanceAnnotations(ODataResource oDataResource, IODataInstanceAnnotationContainer instanceAnnotationContainer, 
-            ODataDeserializerProvider deserializerProvider, ODataDeserializerContext readContext)
+        internal static void SetInstanceAnnotations(
+            ODataResourceBase oDataResource,
+            IODataInstanceAnnotationContainer instanceAnnotationContainer,
+            IODataInstanceAnnotationContainer transientAnnotationContainer,
+            ODataDeserializerProvider deserializerProvider,
+            ODataDeserializerContext readContext)
         {
-            if(oDataResource.InstanceAnnotations != null)
+            if (oDataResource.InstanceAnnotations != null)
```

Comment:
This check is superfluous. The `InstanceAnnoations` property cannot return null.
See
1) https://github.com/OData/odata.net/blob/b37a90e32319636c368a2ce9e5ba2cf39cb65582/src/Microsoft.OData.Core/ODataResource.cs#L210
2) https://github.com/OData/odata.net/blob/b37a90e32319636c368a2ce9e5ba2cf39cb65582/src/Microsoft.OData.Core/ODataAnnotatable.cs#L33